app/controllers/assignments_controller.rb

```diff
@@ -110,8 +111,8 @@ def destroy
       confirm_change(@assignment)
       redirect_to roster_assignments_path(@roster)
     else
-      flash[:errors] = t('.not_an_admin')
-      redirect_to edit_roster_assignment_path(@roster, @assignment)
+      flash.now[:errors] = t('.not_an_admin')
+      render :edit
```

Comment:
```suggestion
      flash[:errors] = t('.not_an_admin')
      redirect_to edit_roster_assignment_path(@roster, @assignment)
```

This one was fine actually - `destroy` actions don't usually have a _dedicated_ form page or form fields so they'll _generally_ always redirect back to where they came from.